desktop/src/main/kotlin/io/spine/examples/pingh/desktop/Login.kt

```diff
@@ -0,0 +1,334 @@
+            ErrorMesage(isError.value)
+        }
+    }
+}
+
+/**
+ * Displays a container for the text field.
+ *
+ * @param value the current value of the input field.
+ * @param textField the composable function that displays the content of an input field.
+ * @param border the border of this input.
+ */
+@Composable
+private fun InputContainer(
+    value: String,
+    textField: @Composable () -> Unit,
+    border: BorderStroke
+) {
+    Row(
+        modifier = Modifier
+            .width(180.dp)
+            .height(40.dp)
+            .border(border = border, shape = MaterialTheme.shapes.medium)
+            .background(
+                color = MaterialTheme.colorScheme.secondary,
+                shape = MaterialTheme.shapes.medium
+            )
+            .padding(horizontal = 10.dp, vertical = 3.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Box(
+            modifier = Modifier
+                .weight(1f)
+                .fillMaxWidth(),
+            contentAlignment = Alignment.CenterStart,
+        ) {
+            Placeholder(value.isEmpty())
+            textField()
+        }
+    }
+}
+
+/**
+ * Displays a label which placed on the top border of the text field.
+ *
+ * @param color the color of this label.
+ */
+@Composable
+private fun Label(color: Color) {
+    Box(
+        modifier = Modifier
+            .width(90.dp)
+            .height(10.dp)
+            .absoluteOffset(x = 10.dp, y = (-5).dp)
+    ) {
+        Text(
+            text = "GitHub username",
+            modifier = Modifier
+                .fillMaxSize()
+                .background(MaterialTheme.colorScheme.secondary),
+            color = color,
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.bodyMedium
+        )
+    }
+}
+
+/**
+ * Displays a placeholder which placed inside the text field.
+ *
+ * @param isShown indicates whether the placeholder is displayed.
+ */
+@Composable
+private fun Placeholder(isShown: Boolean) {
+    if (isShown) {
+        Text(
+            text = "john-smith",
+            color = MaterialTheme.colorScheme.onBackground,
+            style = MaterialTheme.typography.bodyMedium
+        )
+    }
+}
+
+/**
+ * Displays an error message which is placed below the text field.
+ *
+ * @param isShown indicates whether the error message is displayed.
+ */
+@Composable
+private fun ErrorMesage(isShown: Boolean) {
+    if (isShown) {
+        Text(
+            text = "Enter a valid GitHub username.",
+            modifier = Modifier
+                .width(155.dp)
+                .height(30.dp)
+                .absoluteOffset(x = 15.dp, y = 44.dp),
+            color = MaterialTheme.colorScheme.error,
+            style = MaterialTheme.typography.bodySmall
+        )
+    }
+}
+
+/**
+ * Displays a button on the login form.
+ *
+ * @param enabled controls the enabled state of this button.
+ * If `false`, the button cannot be pressed.
```

Comment:
I think this line is not aligned properly. Please have a look throughout the Spine codebase in Kotlin, for examples.